### PR TITLE
Battery UI fixes

### LINKED ISSUE #798: Conditional battery inputs in UI for BTM, FOM, standalone, PV, Generic configurations
## 1. Battery Voltage UI

Battery voltage model option `batt_voltage_choice` and voltage table `batt_voltage_matrix` does not update when battery type changed. For example:

1. Open default PV Battery / Single Owner case.
2. Change voltage model from electrochemical model to voltage table.
3. Change battery type to a vanadium flow battery.

## 2. Standalone Battery

Disable "Minimum time at charge state" under Charge Limits and Priority?

Are all dispatch options available for standalone battery? If not, for now, use msgbox() in UI when user chooses an invalid option for now.

## 3. Generic Battery with Custom Weather File

UI callback error for Generic Battery / Single Owner with **Automated dispatch** and **Look ahead to custom weather file**. Should 1) display msgbox() with "Look ahead to custom weather file is only available for system systems with a photovoltaic array." 2) Disable `batt_select_weather_file_forecast` and `batt_user_specified_weather_file` Browse button and weather file input field.

```
Could not evaluate callback function:batt_select_weather_file_forecast->on_change
[27]: eval error in statement list
[27]: error inside function call
[30]: eval error in statement list
```

## 4. Generic Battery with PV Smoothing

For generic battery, msgbox() when PV smoothing selected, or allow when generation profile option enables with 15-minute or less time step data.

## 5. FOM Automated Dispatch with Specify IRR

SSC should throw an error for this option. Until we make that change, show message on both Battery Dispatch and Revenue (or Financial Parameters for partnership flip and sale leaseback) when user chooses automated dispatch option while in Specify IRR mode (or chooses Specify IRR while in Automatic dispatch mode). Consider using yesno() instead of msgbox() to prompt user to automatically change mode from Specify IRR to Specify PPA price.

### PR DESCRIPTION
**Pull request by mjprilliman**
## Description

- Disable minimum time at charge state for Standalone Battery (unclear why)
- Generic Battery disabling for Automated dispatch custom weather file (similar to Standalone battery)
- Generic Battery msgbox for PVSmoothing
- 
Fixes #798 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

If you have added a new compute module in a SSC pull request related to this one, be sure to check the [Process Requirements](https://github.com/NREL/SAM/wiki/Compute-modules-in-SAM).

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by cpaulgilman — CHANGES_REQUESTED** [review]
I revised the UI message text and set them up to display on_change and on_load.

I'm getting callback errors "attempting to set 'batt_C_rate' (number) to <null>" and sometimes a crash when I change battery type and choose replace defaults option for LFP/Graphite, LMO/LTO, Lead Acid...

File: deploy/runtime/ui/Battery Model.txt [hunk 1]
```diff
@@ -503,25 +503,23 @@ function set_batt_defaults_from_data(defs)
 		value('LeadAcid_tn',cell_params.leadacid_tn);
 	}
 	
-	if (cell_params.voltage_choice == 0) {
```
Comment (dguittet) [inline deploy/runtime/ui/Battery Model.txt:506]: This `if` statement needs to be reinstated. The JSON files do not contain both voltage table data and electrochemical data.

**Review by cpaulgilman — APPROVED** [review]
Looks good after my latest commit to re-enable minimum time at charge state per @brtietz comment, fixes to use consistent specify ppa price/irr messages for standalone and non-standalone FOM batteries, and minor line spacing in message boxes.

**Review by dguittet — APPROVED** [review]

**Comment by mjprilliman** [comment]
Test results were the same outside of ETES (is there variation in the optimizer?). Merging this PR.

**Pull request merged**